server-spi-private/src/main/java/org/keycloak/cookie/CookieType.java

```diff
@@ -29,12 +29,12 @@ public final class CookieType {
             .build();
 
     public static final CookieType LOCALE = CookieType.create("KEYCLOAK_LOCALE")
-            .scope(CookieScope.LEGACY)
+            .scope(CookieScope.FEDERATION)
             .defaultMaxAge(CookieMaxAge.SESSION)
             .build();
 
     public static final CookieType LOGIN_HINT = CookieType.create("KEYCLOAK_REMEMBER_ME")
-            .scope(CookieScope.LEGACY)
+            .scope(CookieScope.FEDERATION)
```

Comment:
Why do these have to be federated? Are they used across domains as well, and in which context?

Reply:
After a redirect to the login pages they wouldn't be available otherwise; locale is used to set the initial locale for a login page when the user is not known, and remember me is just used to populate the username field with the previously used username when remember me was enabled.